src/mono/wasm/test-main.js

```diff
@@ -290,9 +293,10 @@ function configureRuntime(dotnet, runArgs) {
         }
     }
     if (ENVIRONMENT_IS_WEB) {
-        if (runArgs.memorySnapshot) {
+        if (runArgs.memorySnapshot)
             dotnet.withStartupMemoryCache(true);
-        }
+        if (runArgs.interpreterPgo)
```

Comment:
similar to `memorySnapshot` the `interpreterPgo` needs multiple runs to take any effect.
For `memorySnapshot` this script would do  "dryRun" to create such run. But it would NOT execute any user code and will be stopped very early, because that's enough.

I think that means that in a unit test, `interpreterPgo` would never have chance to store the stats and subsequenty real run would not have it available.

I think we have following options
1) don't test interpreterPgo in unit tests, but find another way
2) make `dryRun` to run longer
3) make `dryRun` to run to store stats much earlier, together with memory snapshot (but there was not much managed code running at that point)
4) use some pre-loaded stats. Such stats would not match the exact hash of the binaries, tho

Reply:
I don't think it's possible to exercise this in traditional unit tests, but we could probably exercise it in a WBT test of some sort.